spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/kafka/KafkaPropertiesTests.java

```diff
@@ -0,0 +1,188 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.kafka.listener.AbstractMessageListenerContainer.AckMode;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+/**
+ * @author Gary Russell
+ * @since 1.5
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.NONE, properties = {
+	"spring.kafka.bootstrap-servers=foo:1234",
+	"spring.kafka.clientId=cid",
+	"spring.kafka.ssl-key-password=p1",
+	"spring.kafka.ssl-keystore-location=classpath:ksLoc",
+	"spring.kafka.ssl-keystore-password=p2",
+	"spring.kafka.ssl-truststore-location=classpath:tsLoc",
+	"spring.kafka.ssl-truststore-password=p3",
+
+	"spring.kafka.consumer.auto-commit-interval-ms=123",
+	"spring.kafka.consumer.auto-offset-reset=earliest",
+	"spring.kafka.consumer.client-id=ccid", // test override common
+	"spring.kafka.consumer.enable-auto-commit=false",
+	"spring.kafka.consumer.fetch-max-wait-ms=456",
+	"spring.kafka.consumer.fetch-min-bytes=789",
+	"spring.kafka.consumer.group-id=bar",
+	"spring.kafka.consumer.heartbeat-interval-ms=234",
+	"spring.kafka.consumer.key-deserializer = org.apache.kafka.common.serialization.LongDeserializer",
+	"spring.kafka.consumer.value-deserializer = org.apache.kafka.common.serialization.IntegerDeserializer",
+
+	"spring.kafka.producer.acks=all",
+	"spring.kafka.producer.batch-size=20",
+	"spring.kafka.producer.bootstrap-servers=bar:1234", // test override common
+	"spring.kafka.producer.buffer-memory=12345",
+	"spring.kafka.producer.compression-type=gzip",
+	"spring.kafka.producer.key-serializer=org.apache.kafka.common.serialization.LongSerializer",
+	"spring.kafka.producer.retries=2",
+	"spring.kafka.producer.ssl-key-password=p4",
+	"spring.kafka.producer.ssl-keystore-location=classpath:ksLocP",
+	"spring.kafka.producer.ssl-keystore-password=p5",
+	"spring.kafka.producer.ssl-truststore-location=classpath:tsLocP",
+	"spring.kafka.producer.ssl-truststore-password=p6",
+	"spring.kafka.producer.value-serializer=org.apache.kafka.common.serialization.IntegerSerializer",
+
+	"spring.kafka.template.default-topic=testTopic",
+
+	"spring.kafka.listener.ack-mode=MANUAL",
+	"spring.kafka.listener.ack-count=123",
+	"spring.kafka.listener.ack-time=456",
+	"spring.kafka.listener.concurrency=3",
+	"spring.kafka.listener.poll-timeout=2000"
+	})
+public class KafkaPropertiesTests {
```

Comment:
Such `Properties` object are rather tested as simple unit test, using `EnvironmentContextTestUtils` to bind the relevant keys.